src/arviz_plots/plots/pair_plot.py

```diff
@@ -0,0 +1,495 @@
+    filter_vars: {None, “like”, “regex”}, default None
+        If None (default), interpret var_names as the real variables names.
+        If “like”, interpret var_names as substrings of the real variables names.
+        If “regex”, interpret var_names as regular expressions on the real variables names.
+    group : str, default "posterior"
+        Group to use for plotting. Defaults to "posterior".
+    coords : mapping, optional
+        Coordinates to use for plotting.
+    sample_dims : iterable, optional
+        Dimensions to reduce unless mapped to an aesthetic.
+        Defaults to ``rcParams["data.sample_dims"]``
+    marginal : bool, default True
+        Whether to plot marginal distributions on the diagonal.
+    marginal_kind : {"kde", "hist", "ecdf"}, optional
+        How to represent the marginal density.
+        Defaults to ``rcParams["plot.density_kind"]``
+    triangle : {"both", "upper", "lower"}, Defaults to "both"
+        Which triangle of the pair plot to plot.
+    plot_matrix : PlotMatrix, optional
+    backend : {"matplotlib", "bokeh", "plotly", "none"}, optional
+        Plotting backend to use. Defaults to ``rcParams["plot.backend"]``
+    labeller : labeller, optional
+    aes_by_visuals : mapping, optional
+        Mapping of visuals to aesthetics that should use their mapping in `plot_matrix`
+        when plotted. Valid keys are the same as for `visuals`.
+        By default, there are no aesthetic mappings at all
+    visuals : mapping of {str : mapping or False}, optional
+        Valid keys are:
+
+        * scatter -> passed to :func:`~.visuals.scatter_couple`
+        * divergence -> passed to :func:`~.visuals.scatter_couple`. Defaults to False.
+        * dist -> depending on the value of `marginal_kind` passed to:
+
+          * "kde" -> passed to :func:`~arviz_plots.visuals.line_xy`
+          * "ecdf" -> passed to :func:`~arviz_plots.visuals.ecdf_line`
+          * "hist" -> passed to :func: `~arviz_plots.visuals.hist`
+
+        * credible_interval -> passed to :func:`~arviz_plots.visuals.line_x`
+        * point_estimate -> passed to :func:`~arviz_plots.visuals.scatter_x`
+        * point_estimate_text -> passed to :func:`~arviz_plots.visuals.point_estimate_text`
+        * label -> passed to :func:`~.visuals.label_plot`
+        * xlabel -> passed to :func:`~.visuals.labelled_x`
+        * ylabel -> passed to :func:`~.visuals.labelled_y`
+        * diag_xlabel -> passed to :func:`~.visuals.labelled_x` for the diagonal plots
+        * remove_axis -> not passed anywhere, can only be ``True`` to  call this function
+
+    stats : mapping, optional
+        Valid keys are:
+
+        * dist -> passed to kde, ecdf, ...
+        * credible_interval -> passed to eti or hdi
+        * point_estimate -> passed to mean, median or mode
+
+    **pc_kwargs
+        Passed to :class:`arviz_plots.PlotMatrix`
+
+
+    Returns
+    -------
+    PlotMatrix
+
+    Examples
+    --------
+    Default plot_pair
```

Comment:
The examples should focus on illustrating the behaviour that is specific to `plot_matrix` and can be inferred/learned from the general tutorial. At the same time doing that helps clarify the behaviour of unique/complicated arguments.

In this case I would generate multiple plots with `plot_pair(dt, var_names=["mu", "tau"], marginal=, triangle=)` for different values of marginal and triangle to show the less common/potentially surprising cases. For example, with lower triangle only, the behaviour of `marginal` True/False is quite simple to explain, if the diagonal is free the labels with variable names go there, otherwise they go at the left/bottom of the figure. However, the cases with upper are not as polished (which is ok) but should have extensive docs to minimize issues and surprises